### PR TITLE
fix: Propagate error when bridge account already exists for IBAN

### PR DESCRIPTION
**Pull request by panosfilianos**
Also adds example of Respone interface type

### CODE REVIEW

File: src/app/api/bridge/external-account/create-external-account/route.ts [hunk 1]
```diff
@@ -68,6 +68,26 @@ export async function POST(request: NextRequest) {
...
+                // sending back error responses is not currently common across app 
+                // in how we deliver errors from backend -> frontend
+                // sends back an object like:
+                // {
+                //     id: '4c537540-80bf-41dd-a528-dbe39a4',
+                //     code: 'duplicate_external_account',
+                //     message: 'An external account with the same information has already been added for this customer'
+                //   }
+                //
+                // TODO: standardize error responses across backend
```
Comment (Hugo0) [inline src/app/api/bridge/external-account/create-external-account/route.ts:83]: praise: nice comment!

File: src/components/Global/LinkAccountComponent/index.tsx [hunk 2]
```diff
@@ -193,7 +206,7 @@ export const GlobaLinkAccountComponent = ({ accountNumber, onCompleted }: IGloba
             onCompleted ? onCompleted() : setCompletedLinking(true)
         } catch (error) {
             console.error('Error during the submission process:', error)
-            setErrorState({ showError: true, errorMessage: 'An error occurred. Please try again later' })
+            setErrorState({ showError: true, errorMessage:  String(error)})
```
Comment (Hugo0) [inline src/components/Global/LinkAccountComponent/index.tsx:209]: suggestion: Shouldn't we use error.message? String(error) is gonna be gibberish

File: src/utils/cashout.utils.ts [hunk 3]
```diff
@@ -187,13 +187,36 @@ export async function createExternalAccount(
             }),
         })
 
+        const data = await response.json()
+
         if (!response.ok) {
+            if (data.code && data.code == 'duplicate_external_account') {
```
[the remark this answers is not in the document]
Comment (Hugo0) [reply src/utils/cashout.utils.ts:193]: nit: this is a nitpick and wont change anth here, but generally good advice

**Review by Hugo0 — APPROVED** [review]
@panosfilianos

File: src/components/Global/LinkAccountComponent/index.tsx (hunk 2 above)
Comment (panosfilianos) [reply src/components/Global/LinkAccountComponent/index.tsx:209]: There is string repr that makes the conversion for the Error type

**Pull request merged**